### PR TITLE
Add support for keyUp and keyPress events

### PR DESCRIPTION
**Pull request by tbarlow12**
- Still needs test, opening up to get feedback on approach
- Removed unused keydown package

### CODE REVIEW

File: src/react/components/common/keyboardBinding/keyboardBinding.test.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ describe("Keyboard Binding Component", () => {
 
     const defaultProps: IKeyboardBindingProps = {
         accelerator: "Ctrl+1",
-        onKeyDown: onKeyDownHandler,
+        onKeyEvent: onKeyDownHandler,
```
Comment (mydiemho) [inline src/react/components/common/keyboardBinding/keyboardBinding.test.tsx]: maybe this should also be rename `onKeyEventHandler`

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.ts [hunk 2]
```diff
@@ -4,7 +4,18 @@ import Guard from "../../../../common/guard";
...
+    [keyEventType: string]: {
+        [key: string]: KeyboardEventHandler[],
+    };
+}
+
+/**
+ * Types of Key events supported by registration manager
+ */
+export enum KeyEventType {
+    KeyDown = "keydown",
```
Comment (mydiemho) [inline src/react/components/common/keyboardManager/keyboardRegistrationManager.ts]: Nit: camelCase for value?

File: src/react/components/common/keyboardBinding/keyboardBinding.test.tsx (hunk 1 above)
Comment (tbarlow12) [reply src/react/components/common/keyboardBinding/keyboardBinding.test.tsx]: I think I'll keep it without the `Handler` just to keep it consistent with other event handlers like `onClick`, `onCancel`, etc.

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.ts (hunk 2 above)
Comment (tbarlow12) [reply src/react/components/common/keyboardManager/keyboardRegistrationManager.ts]: I think the event needs to be all lowercase. I was just following the convention that was already here. @wbreza any thoughts?

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.ts [hunk 3]
```diff
@@ -45,23 +54,24 @@ export class KeyboardRegistrationManager {
      * Gets a list of registered event handlers for the specified key code
      * @param keyCode The key code combination, ex) Ctrl+1
      */
-    public getHandlers(keyCode: string) {
+    public getHandlers(keyEventType: KeyEventType, keyCode: string) {
```
Comment (JacopoMangiavacchi) [inline src/react/components/common/keyboardManager/keyboardRegistrationManager.ts]: keyEventType param missing in jdocs

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts [hunk 4]
```diff
@@ -18,11 +19,11 @@ describe("Keyboard Registration Manager", () => {
         const keyCode2 = "Ctrl+S";
         const handler2 = (evt: KeyboardEvent) => null;
 
-        keyboardManager.addHandler(keyCode1, handler1);
-        keyboardManager.addHandler(keyCode2, handler2);
+        keyboardManager.addHandler(KeyEventType.KeyDown , keyCode1, handler1);
```
Comment (elizabethhalper) [inline src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts]: nit: extra space

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts [hunk 5]
```diff
@@ -31,30 +32,55 @@ describe("Keyboard Registration Manager", () => {
         expect(handlers2[0]).toBe(handler2);
     });
 
+    it("can register handlers for same key code and different key event types", () => {
+        const keyCode = "Ctrl+H";
```
Comment (elizabethhalper) [inline src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts]: Is H just random letter you're using to test this or is there a reason for it?

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.ts [hunk 6]
```diff
@@ -1,10 +1,13 @@
 import Guard from "../../../../common/guard";
+import { KeyEventType } from "./keyboardManager";
 
 /**
  * A map of keyboard event registrations
  */
 export interface IKeyboardRegistrations {
-    [key: string]: KeyboardEventHandler[];
+    [keyEventType: string]: {
```
Comment (wbreza) [inline src/react/components/common/keyboardManager/keyboardRegistrationManager.ts]: Should this be a `KeyEventType` ?

File: src/react/components/common/keyboardBinding/keyboardBinding.tsx [hunk 7]
```diff
@@ -1,9 +1,10 @@
-import { KeyboardContext, IKeyboardContext } from "../keyboardManager/keyboardManager";
+import { KeyboardContext, IKeyboardContext, KeyEventType } from "../keyboardManager/keyboardManager";
 import React from "react";
 
 export interface IKeyboardBindingProps {
     accelerator: string;
-    onKeyDown: (evt?: KeyboardEvent) => void;
+    keyEventType: KeyEventType;
```
Comment (wbreza) [inline src/react/components/common/keyboardBinding/keyboardBinding.tsx]: Do you think it makes sense to make the event type optional and default to keydown? or just make it required and explicit?  I have not string preference.

Comment (tbarlow12) [reply src/react/components/common/keyboardBinding/keyboardBinding.tsx]: Yeah, that actually does make sense. I'll make the change

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.ts (hunk 6 above)
Comment (tbarlow12) [reply src/react/components/common/keyboardManager/keyboardRegistrationManager.ts]: I wanted to, but it forces me to use number or string

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts (hunk 5 above)
Comment (tbarlow12) [reply src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts]: Yep, just random :)

File: src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts (hunk 4 above)
Comment (tbarlow12) [reply src/react/components/common/keyboardManager/keyboardRegistrationManager.test.ts]: Thanks

**Pull request merged**